sys/include/shell.h

```diff
@@ -283,15 +283,14 @@ int shell_parse_file(const shell_command_t *commands,
  * ```
  */
 #define SHELL_COMMAND(cmd, help, func) \
-    XFA_USE_CONST(shell_command_xfa_t*, shell_commands_xfa); \
+    XFA_USE_CONST(shell_command_xfa_t, shell_commands_xfa_v2); \
```

Comment:
Why do we need to rename this? I would have expected this name to be an implementation detail.

Reply:
It is an implementation detail, but the implementation of shell XFA is split across two repos: This repo contains the C implementation, https://github.com/RIOT-OS/rust-riot-wrappers contains the rust implementation.

If the rust impl would live here as well, I would have just adapted both in lock-step and kept the symbol name. But with the current state of affair, there is a benefit to be able to implement the rust side in a way that it will work with both variants of RIOT. (E.g. so that we can update the rust side while using a stable release on the C side.) See https://github.com/RIOT-OS/rust-riot-wrappers/pull/134 for why this is beneficial for the rust side.

Reply:
Alright, thanks for the explanation, makes sense!